### PR TITLE
gnupg url and mirror updates

### PR DESCRIPTION
**Pull request by vszakats**
* replace ftp with secure protocol
* sync mirrors across gnupg projects
* replace `mirrors.kernel.org` with `mirrorservice.org`
* consistent use of `gnupg.org` (for URLs) vs. `www.gnupg.org` (for homepages)
* remove insecure mirrors
* homepage updates and cleanups
* minor sync of build options formatting

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Could you take a look at the CI failures? Thanks!

**Comment by vszakats** [comment]
It fails in `dirmngr` which was not updated in this PR, and just passed tests successfully an hour ago in a separate [PR](https://github.com/Homebrew/homebrew/pull/45817).

```
==> dirmngr-client --help
dyld: Library not loaded: /usr/local/lib/libassuan.0.dylib
  Referenced from: /usr/local/bin/dirmngr-client
  Reason: image not found
Error: dirmngr: failed
Failed executing: dirmngr-client --help
```

File: Library/Formula/libksba.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Libksba < Formula
   desc "X.509 and CMS library"
   homepage "https://www.gnupg.org/related_software/libksba/index.en.html"
-  url "ftp://ftp.gnupg.org/gcrypt/libksba/libksba-1.3.3.tar.bz2"
```
Comment (DomT4) [inline Library/Formula/libksba.rb]: Please keep some form of official upstream URL in place here.

Comment (vszakats) [inline Library/Formula/libksba.rb]: Tthat was a mistake, thanks for noticing.

Comment (vszakats) [inline Library/Formula/libksba.rb]: Fixed! (Is there a way to squash two non-consecutive commits?)

Comment (mikemcquaid) [inline Library/Formula/libksba.rb]: `git rebase --interactive` and cut-and-paste the lines to a different order.

Comment (DomT4) [inline Library/Formula/libksba.rb]: Yeah. If you `git rebase -i HEAD~12` and then just drag/drop your commits into the order desired and squash as per normal.

Comment (DomT4) [inline Library/Formula/libksba.rb]: Oops, Mike beat me to it :smile_cat:

Comment (vszakats) [inline Library/Formula/libksba.rb]: Thanks to both of you! Got it the way you described 👍🏻

Comment (vszakats) [inline Library/Formula/libksba.rb]: Did some `gnupg.org` (URLs) vs. `www.gnupg.org` (homepages) inconsistency fixup. They may be further synced by using the latter for URLs as well.

**Comment by vszakats** [comment]
Question: Shouldn't be all additional insecure mirrors removed? The canonical and `mirrorservice.org` URLs would still remain. (Having an insecure mirror can be exploited in theory.)

**Comment by DomT4** [comment]
Feel free to kill anything that isn't SSL/TLS, IMO, presuming we have one good `url` and one good `mirror`.

**Comment by vszakats** [comment]
Cool, will do!

**Comment by vszakats** [comment]
Done.

**Comment by vszakats** [comment]
What does the new result `UNSTABLE` mean?
```
==> brew cleanup --prune=7
Recording test results
Build step 'Publish JUnit test result report' changed build result to UNSTABLE
Archiving artifacts
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: UNSTABLE
```

**Comment by DomT4** [comment]
It's not new. Just doesn't appear very often. Official definition from Jenkins is:

> A build is unstable if it was built successfully and one or more publishers report it unstable. For example if the JUnit publisher is configured and a test fails then the build will be marked unstable.

**Comment by DomT4** [comment]
In Homebrew terms, you can usually read `UNSTABLE` as *"Something broke that wasn't my fault and can probably be ignored by the maintainers"*, to be honest.

**Comment by vszakats** [comment]
Thanks @DomT4. I'll make some more minor cleanups, but still nothing that should cause any change in the actual build processes.

**Comment by vszakats** [comment]
Okay, finished with that, too.

**Comment by DomT4** [comment]
Merged in a whole bunch of commits. Thank you @vszakats :smiley_cat:

**Comment by vszakats** [comment]
Thank you so much @DomT4!

**Pull request closed without merging**